### PR TITLE
(refonte) Preview modal

### PR DESCRIPTION
**Pull request by gmougeolle**
**Layout**

- Dataset : private author / public service disclaimer is now below the title
- Dataset : Author name is now displayed on community resources
- Toolbar : deleted the unneeded toolbar block
- Resource-card : Restored extra buttons hook
- Resource-card : preview modal implemented
- Base : removed notification zone

**Ui-kit**

- New xxs spacing for very smol spacing needs
- Markdown : new markdown component (and corresponding doc) for styling markdown-generated markup
- Modal : new modal modular component & doc

### CODE REVIEW

File: theme/js/components/discussions/thread.vue [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+</template>
+
+<script>
+import ThreadReply from "./thread-reply.vue";
+import Avatar from "./avatar.vue";
+import LinkIcon from "svg/permalink.svg";
+import dayjs from "dayjs";
+import "dayjs/locale/fr";
+
+dayjs.locale("fr");
```
Comment (abulte) [inline theme/js/components/discussions/thread.vue:77]: This should be configurable. We currently have a JS variable called `uData.config.lang` that might do the trick.

File: theme/js/components/discussions/threads-create.vue [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    };
+  },
+  props: {
+    subjectId: String,
+    subjectClass: String,
+    onSubmit: Function,
+  },
+  methods: {
+    displayForm: function() {
+       this.$auth('You need to be logged in to start a discussion.');
```
Comment (abulte) [inline theme/js/components/discussions/threads-create.vue:44]: Translation? Or maybe you're keeping this for later (but a XXX wouldn't hurt I guess).

File: theme/js/components/discussions/threads.vue [hunk 3]
```diff
@@ -0,0 +1,154 @@
+<template>
+  <section class="discussions-wrapper">
+    <transition mode="out-in">
+      <div v-if="loading" key="loader">
+        <vue-content-loading
```
Comment (abulte) [inline theme/js/components/discussions/threads.vue:5]: not worth making a component?

File: theme/js/components/discussions/threads.vue [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+          <rect x="98" y="46" rx="3" ry="3" width="250" height="5"></rect>
+          <rect x="98" y="59" rx="3" ry="3" width="250" height="5"></rect>
+          <rect x="26" y="100" rx="5" ry="5" width="60" height="60"></rect>
+          <rect x="97" y="100" rx="3" ry="3" width="120" height="8"></rect>
+          <rect x="98" y="118" rx="3" ry="3" width="250" height="5"></rect>
+          <rect x="98" y="131" rx="3" ry="3" width="250" height="5"></rect>
+        </vue-content-loading>
+      </div>
+      <div v-if="!loading" ref="top" key="top">
+        Trié par : {{ current_sort.name }}
```
Comment (abulte) [inline theme/js/components/discussions/threads.vue:25]: Okay, translation everywhere ;-) Let me know how you want to deal with this.

File: theme/js/config.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
+/*
+ * Parse the page html headers to extract some informations
```
Comment (abulte) [inline theme/js/config.js:2]: 💪

File: theme/js/plugins/api.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import axios from "axios";
+import config from "../config";
+
+//Instanciate axios with base URL from config
+const api = axios.create({
+  baseURL: config.api_root,
+});
+
+//TODO, probably : add interceptors to add CSRF info and credentials
```
Comment (abulte) [inline theme/js/plugins/api.js:9]: I don't think so, credentials are handled through sessions and no CSRF on API endpoint (not sure).

**Review by abulte — COMMENTED** [review]
Awesome! A few comments.

**Comment by gmougeolle** [comment]
Dupe of #510

**Pull request closed without merging**